pkg/acs/calacs/acsccd/doccd.c

```diff
@@ -143,278 +171,582 @@ int DoCCD (ACSInfo *acs_info) {
+    if (acs_info->blevcorr == PERFORM) {
+
+       /* This block handles only the WFC detector */
+       if (acs_info->detector == WFC_CCD_DETECTOR) {
+
+          /* If this is subarray data, determine if it is supported for the bias shift 
+             correction.  Compare the aperture name against the array of supported
+             apertures.  Use the aperture name and not the date as the discriminant 
+             as the date is fuzzy - some supported subarrays were acquired before the 
+             official name change, and some old style subarrays were acquired after the 
+             change.  This scheme works fine for non-polarization and non-ramp data as the
+             "*-2K" apertures have both physical and virtual overscan.  These are needed
+             to compute the "fat zero" (aka magic square) value needed for the bias
+             shift correction.  The "*-1K and *-512" subarrays do not have virtual 
+             overscan which is needed at this time for fat zero.
+
+             For polarization and ramp subarrays, the aperture names did not change.  For 
+             this data, the dimensions of the image also have to be checked to determine 
+             if there are both physical and virtual overscan sections so the bias shift 
+             correction can be applied.
+          */
+          if (acs_info->subarray == YES) {
+             {unsigned int i;
+             for (i = 0; i < numSupportedSubApertures; i++) {
+                 if (strcmp(acs_info->aperture, subApertures[i]) == 0) {
+
+                    /* If this is polarization or ramp data, then the size of the science
+                       array must also be checked to make sure there is physical and
+                       virtual overscan data.  This does not need to be checked explicitly
+                       for non-polarization or non-ramp data, but the comparison works for 
+                       this data too.  Only need to check one virtOverscan value in the array. 
+
+                       NOTE:Just because the aperture is supported does not mean the bias
+                       shift correction will be applied.  The DS_int and gain criteria also
+                       have to be satisfied.
+                    */
+                    if (virtOverscan[0]) {
+                       isSupportedSubAperture = YES;
+                       trlmessage("This subarray data is supported for the bias shift correction.");
+                    }
+                 }
+             }}
+          }
+
+          /* Only Pre-SM4 WFC data - this is the original bias level subtraction */
+          if (acs_info->expstart < SM4MJD) {
+
+             {unsigned int i;
+             for (i = 0; i < acs_info->nimsets; i++) {
+
+                 /* The variable done is set to the results of FindOver indicating there is
+                    physical overscan. */
+                 done = overscan[i];
```

Comment:
Making this a  note since you didn't edit ``doBlev()``. It's better to  not overload params, i.e. separate those that are input and those that are output. In this case it's ``done``. This should either be past in (As const) or a ptr so that it can be modified as output, but never both.